WordPress/Classes/Services/PostCoordinator.swift

```diff
@@ -151,6 +151,52 @@ class PostCoordinator: NSObject {
         AppRatingUtility.shared.incrementSignificantEvent()
     }
 
+    /// Uploads the changes made to the post to the server.
+    ///
+    /// - warning: Work-in-progress (kahu-offline-mode)
+    @discardableResult @MainActor
+    func _update(_ post: AbstractPost, changes: RemotePostUpdateParameters? = nil) async throws -> AbstractPost {
+        let post = post.original ?? post
+        do {
+            let isExistingPost = post.hasRemote()
+            // TODO: Set overwrite to false once conflict resolution support is added
+            try await PostRepository()._save(post, changes: changes, overwrite: true)
+            show(PostCoordinator.makeUploadSuccessNotice(for: post, isExistingPost: isExistingPost))
+            return post
+        } catch {
+            handleError(error, for: post)
+            throw error
+        }
+    }
+
+    private func handleError(_ error: Error, for post: AbstractPost) {
+        guard let topViewController = UIApplication.shared.mainWindow?.topmostPresentedViewController else {
+            return
+        }
+        let alert = UIAlertController(title: Strings.uploadFailed, message: error.localizedDescription, preferredStyle: .alert)
```

Comment:
This is one change from the spec. I used `UIAlert` because:
- It stays on the screen so you can't miss it
- It can display long error messages
- It can have recovery actions
- Apple uses alerts for errors